src/EditorFeatures/VisualBasic/QuickInfo/SemanticQuickInfoProvider.vb

```diff
@@ -92,6 +93,19 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.QuickInfo
             Return Await MyBase.BuildContentAsync(document, token, cancellationToken).ConfigureAwait(False)
         End Function
 
+        ''' <summary>
+        ''' If the token is a 'Sub' or 'Function' in a lambda, returns the syntax for the whole lambda
+        ''' </summary>
+        Protected Overrides Function GetBindableNodeForTokenIndicatingLambda(token As SyntaxToken, <Out> ByRef found As SyntaxNode) As Boolean
+            If token.IsKind(SyntaxKind.SubKeyword, SyntaxKind.FunctionKeyword) AndAlso token.Parent.IsKind(SyntaxKind.SubLambdaHeader, SyntaxKind.FunctionLambdaHeader) Then
```

Comment:
to verify: this works for single and multi-line VB lambdas?

Reply:
Yes. Here's what the syntax tree looks like (for Sub):

[Multi-line Sub](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBsQDdhJiaMA+AwitFAASECwAUGXWQMoCuwZAsgBQAeZAguQEkAdvADmiAJQ16MsgBEAlmDIQ+5BgE9YcMADpeAYxgKA9kLIBeRiw5Tashzys8A1GQCM0h2QCiQgCbWwF50foHMwdRhFCRQUEA=)
![image](https://user-images.githubusercontent.com/12466233/35416433-ec0d6d8e-01dd-11e8-92e5-062178443d25.png)

[Single-line Sub](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBsQDdhJiaMA+AwitFAASECwAUGXWQMoCuwZAsgBQAeZAguQEkAdvADmiAJQ16MsgBEAlmDIQ+5BgE9YcMADpeAYxgKA9kLIBeRiw4SyPKzwDUZAIw1p9AKJCAJteAaH39iUiA=)

![image](https://user-images.githubusercontent.com/12466233/35416442-f4ca446a-01dd-11e8-98e8-cd3b2594e3a5.png)

Also, I have tests for both multi-line and single-line:
```VB

        <Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)>
        <WorkItem(23307, "https://github.com/dotnet/roslyn/issues/23307")>
        Public Async Function TestQuickInfoCaptures() As Task
            Await TestAsync("
Class C
    Sub M(x As Integer)
        Dim a As System.Action = Sub$$()
            x = x + 1
        End Sub
    End Sub
End Class
",
                Captures($"{vbCrLf}{WorkspacesResources.Variables_captured_colon} x"))
        End Function

        <Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)>
        <WorkItem(23307, "https://github.com/dotnet/roslyn/issues/23307")>
        Public Async Function TestQuickInfoCaptures2() As Task
            Await TestAsync("
Class C
    Sub M(x As Integer)
        Dim a As System.Action = S$$ub() x = x + 1
    End Sub
End Class
",
                Captures($"{vbCrLf}{WorkspacesResources.Variables_captured_colon} x"))
        End Function
```